src/transactions/feed/queryHelper.ts

```diff
@@ -101,10 +105,24 @@ export function useFetchTransactions(): QueryHookResult {
 
       if (isPollResult && returnedTransactions.length) {
         // We store the first page in redux to show them to the users when they open the app.
+        // Filter out now empty transactions to avoid redux issues
         const nonEmptyTransactions = returnedTransactions.filter(
           (returnedTransaction) => !isEmpty(returnedTransaction)
         )
-        dispatch(updateTransactions(nonEmptyTransactions))
+        // Compare the new tx hashes with the ones we already have in redux
+        for (let i = 0; i < nonEmptyTransactions.length; i++) {
```

Comment:
I used the for loop with a break to exit the loop as soon as we got a hit, instead of iterating through all the items and determining if we have a new transaction. If nonEmptyTransactions has a new transaction it would likely be toward the start of the array and we can avoid checking all the transactions once we find a new one.